pkg/trace/agent/normalizer.go

```diff
@@ -54,6 +55,17 @@ func normalize(ts *info.TagStats, s *pb.Span) error {
 	}
 	s.Service = svc
 
+	if config.HasFeature("component2name") {
```

Comment:
The `config.HasFeature("component2name")` call incurs synchronization upon the `os.Getenv` call and spends some CPU time scanning the value of the DD_APM_FEATURES variable. As this logic is executed once for every single span the agent receives, it can affect the agent's throughput and CPU utilization significantly. Would it make sense to move the `config.HasFeature("component2name")` call off of the critical path by introducing a variable and using it here?

Reply:
NVM, just saw the changes to the `config.HasFeature` implementation

Reply:
All good, I took this feedback into account. It was a bad design previously.